### PR TITLE
feature/CVSB-1038

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

File: src/providers/global/auth.service.ts [hunk 1]
```diff
@@ -1,44 +1,71 @@
...
+  getJWTToken() {
+    return this.jwtToken
+  }
+
+  private setTesterDetails(token) {
+    let decodedToken = this.decodeJWT(token);
+
+    this.testerDetails.testerId = decodedToken['oid'];
+    this.testerDetails.testerName = decodedToken['name'];
+    this.testerDetails.testerEmail = decodedToken['upn'];
```
Comment (mgherceanu) [inline src/providers/global/auth.service.ts]: use enums if possible please

File: src/providers/interceptors/auth.interceptor.ts [hunk 2]
```diff
@@ -1,16 +1,78 @@
...
+        setHeaders: {
+          "Authorization": AUTH_HEADER
+        }
+      });
+    }
+    return request;
+  }
+
+  logoutUser(error?): Observable<any> {
+    return _throw(error || 'Invalid Token');
```
Comment (mgherceanu) [inline src/providers/interceptors/auth.interceptor.ts]: use enums if possible please

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
+  constructor(platform: Platform, statusBar: StatusBar, splashScreen: SplashScreen, private alertCtrl: AlertController, private syncService: SyncService, private authService: AuthService, private mobileAccessibility: MobileAccessibility, private renderer: Renderer2, public storageService: StorageService, public visitService: VisitService, private deeplinks: Deeplinks) {
     platform.ready().then(() => {
       statusBar.overlaysWebView(true);
       statusBar.styleLightContent();
-
-
+      this.authService.login().subscribe(
+        (data: string) => {
+          this.authService.setJWTToken(data);
+          console.log('app login data: ', data)
```
Comment (mgherceanu) [inline src/app/app.component.ts]: remove console

File: src/app/app.component.ts [hunk 4]
```diff
@@ -53,37 +57,34 @@ export class MyApp {
...
   }
 
+  ngAfterViewInit(): void {
+    this.deeplinks.routeWithNavController(this.navElem, {
+      '/home': 'TestStationHomePage',
+    }).subscribe(
+      (match) => {
+        console.log('Successfully routed', match);
+      }, (nomatch) => {
+        console.warn('Unmatched Route', nomatch);
```
Comment (mgherceanu) [inline src/app/app.component.ts]: remove console logs

File: src/providers/interceptors/auth.interceptor.ts (hunk 2 above)
Comment (danieliordache) [reply src/providers/interceptors/auth.interceptor.ts]: done

File: src/app/app.component.ts (hunk 4 above)
Comment (danieliordache) [reply src/app/app.component.ts]: there is not logic attached to it. It is present for the subscription to work

File: src/app/app.enums.ts [hunk 5]
```diff
@@ -6,13 +6,20 @@ export enum STORAGE {
...
   STATE = 'state',
   VISIT = 'visit',
-  TEST_HISTORY = 'test-history'
+  TEST_HISTORY = 'test-history',
+  JWT_TOKEN = 'jwt-token'
 }
 
 export enum PATHS {
+  HEADER = 'Authorization',
+  JWT_START = 'Bearer ',
```
Comment (acozianu) [inline src/app/app.enums.ts]: Please remove

File: src/app/app.enums.ts [hunk 6]
```diff
@@ -195,14 +201,23 @@ export enum EU_VEHICLE_CATEGORY {
   CAT_N1 = 'n1',
   CAT_N2 = 'n2',
   CAT_N3 = 'n3',
-  CAT_01 = 'o1',
-  CAT_02 = 'o2',
-  CAT_03 = 'o3',
-  CAT_04 = 'o4'
+  CAT_01 = '01',
+  CAT_02 = '02',
```
Comment (acozianu) [inline src/app/app.enums.ts]: change to "o2"

File: src/pages/test-station/test-station-home/test-station-home.ts [hunk 7]
```diff
@@ -4,7 +4,9 @@ import { StorageService } from "../../../providers/natives/storage.service";
 import { VisitService } from "../../../providers/visit/visit.service";
 import { LOCAL_STORAGE, STORAGE } from "../../../app/app.enums";
 
-@IonicPage()
+@IonicPage({
+  segment: 'home'
```
Comment (acozianu) [inline src/pages/test-station/test-station-home/test-station-home.ts]: please remove

File: src/pages/test-station/test-station-home/test-station-home.ts [hunk 8]
```diff
@@ -24,7 +26,7 @@ export class TestStationHomePage implements OnInit {
     this.count = 0;
   }
 
-  getStarted(): void {
+  getStarted() {
```
Comment (acozianu) [inline src/pages/test-station/test-station-home/test-station-home.ts]: add type for the method

File: src/app/app.enums.ts (hunk 5 above)
Comment (danieliordache) [reply src/app/app.enums.ts]: done

File: src/app/app.enums.ts (hunk 6 above)
Comment (danieliordache) [reply src/app/app.enums.ts]: done

File: src/pages/test-station/test-station-home/test-station-home.ts (hunk 7 above)
Comment (danieliordache) [reply src/pages/test-station/test-station-home/test-station-home.ts]: done

File: src/pages/test-station/test-station-home/test-station-home.ts (hunk 8 above)
Comment (danieliordache) [reply src/pages/test-station/test-station-home/test-station-home.ts]: done

**Pull request merged**